src/Microsoft.Health.Fhir.Shared.Core/Features/Search/SearchOptionsFactory.cs

```diff
@@ -58,13 +59,22 @@ public SearchOptions Create(string compartmentType, string compartmentId, string
             {
                 if (query.Item1 == KnownQueryParameterNames.ContinuationToken)
                 {
+                    // This is an unreachable case. The mapping of the query parameters makes it so only one continuation token can exist.
                     if (continuationToken != null)
                     {
                         throw new InvalidSearchOperationException(
                             string.Format(Core.Resources.MultipleQueryParametersNotAllowed, KnownQueryParameterNames.ContinuationToken));
                     }
 
-                    continuationToken = query.Item2;
+                    // Checks if the continuation token is base 64 bit encoded. Needed for systems that have cached continuation tokens from before they were encoded.
+                    if (Regex.IsMatch(query.Item2, "^([A-Za-z0-9+/]{4})*([A-Za-z0-9+/]{3}=|[A-Za-z0-9+/]{2}==)?$"))
```

Comment:
>if (Regex.IsMatch(query.Item2, "^([A-Za-z0-9+/]{4})*([A-Za-z0-9+/]{3}=|[A-Za-z0-9+/]{2}==)?$")) [](start = 20, length = 95)

Is there any chance that older continuation tokens which are not base 64 encoded will pass this regex test?  I am not sure what they look like.

Reply:
The unencoded token seems to be JSON formatted, so it will have characters like { that wouldn't show up in a Base 64 encoded string. Here is an example of an old token taken from the debugger: 
"{\"token\":\"+RID:CvwaAKHaQvz9FgAAAAAAAA==#RT:1#TRC:10#FPC:AgEAAAAGAP2WAcD8/w==\",\"range\":{\"min\":\"\",\"max\":\"FF\"}}"

Reply:
Sounds good.

Reply:
We should also use compiled regex (maybe in a static variable) here since it could be a common call